lib/domain/entities/message/sms.ts

```diff
@@ -0,0 +1,113 @@
+import { Message, MessageData } from './message'
+import { normalizeDiacritics } from 'normalize-text'
+import { SMSShortify } from '../../service/smsshortify'
+
+type MessageType = {
+  prefix?: string;
+  text: string;
+  suffix?: string;
+  variables?: Record<string, string>;
+};
+
+type RecipientType = {
+  phone: string;
+};
+
+export type SMSData = {
+  message: MessageType;
+  recipient: RecipientType;
+} & MessageData;
+
+export class SMS implements Message {
+  readonly channel: string = 'sms'
+  readonly externalId?: string
+  private readonly recipient: RecipientType
+  private readonly message: MessageType
+  private readonly shortifyService: SMSShortify
+
+  constructor({ message, recipient, externalId }: Pick<SMSData, 'message' | 'recipient' | 'externalId'>) {
+    this.externalId = externalId
+    this.message = this.normalize(message)
+    this.recipient = recipient
+    this.replaceVariables()
+    this.shortifyService = new SMSShortify({
+      text: this.text,
+      prefix: this.prefix,
+      suffix: this.suffix,
+      link: this.variables?.link
+    })
+  }
+
+  public getMessage(): SMSData {
+    this.format()
+
+    return {
+      externalId: this.externalId,
+      message: {
+        text: this.text
+      },
+      channel: this.channel,
+      recipient: this.recipient
+    }
+  }
+
+  public shortify(char = 160) {
+    const { text, prefix, suffix } = this.shortifyService.execute(char)
+    this.text = text
+    this.prefix = prefix
+    this.suffix = suffix
+  }
+
+  private format() {
+    if (!this.prefix && !this.suffix) return
+    if (this.prefix && this.suffix) this.text = `${this.prefix}: ${this.text} ${this.suffix}`
```

Comment:
```suggestion
    if (this.prefix && this.suffix) this.text = `${this.prefix} ${this.text} ${this.suffix}`
```
O `:` não deveria ser parte do padrão da mensagem, ele poderia ser adicionado por quem está mandando a mensagem.

Eu posso escolher algo como:
`prefix: "Oi José!"` (o separador sendo uma exclamação)
`prefix: "Suave meu mano?"` (o separador sendo uma interrogação)

Ou seja, não acho que decidir que o separador após o prefixo será sempre `:` seja um bom caminho.

Reply:
Boa, eu tava achando meio estranho tbm, vou ajustar!